Mathlib/Analysis/SpecialFunctions/Gamma/Beta.lean

```diff
@@ -458,10 +458,95 @@ theorem Gamma_ne_zero_of_re_pos {s : ℂ} (hs : 0 < re s) : Gamma s ≠ 0 := by
   contrapose! hs
   simpa only [hs, neg_re, ← ofReal_natCast, ofReal_re, neg_nonpos] using Nat.cast_nonneg _
 
+lemma integral_re {X : Type*} [MeasurableSpace X]
+    (μ : Measure X) {f : X → ℂ} (hf : Integrable f μ) :
+    ∫ x, (f x).re ∂μ = (∫x, f x ∂μ).re := _root_.integral_re hf
+
+lemma setIntegral_re {X : Type*} [MeasurableSpace X]
+    (μ : Measure X) {s : Set X} {f : X → ℂ} (hf : IntegrableOn f s μ) :
+    ∫ x in s, (f x).re ∂μ = (∫x in s, f x ∂μ).re := Complex.integral_re _ hf.integrable
+
 end Complex
 
 namespace Real
 
+/-- The `Γ` function (of a real variable `s`). -/
+@[pp_nodot]
+def Beta (a b : ℝ) : ℝ :=
+  (Complex.betaIntegral a b).re
+
+lemma integrable_iff_ofReal {X : Type*} [MeasurableSpace X] {μ : Measure X}
+    {f : X → ℝ} : Integrable f μ ↔ Integrable (fun x ↦ (f x : ℂ)) μ :=
+    ⟨fun hf ↦ hf.ofReal, fun hf ↦ hf.re⟩
+
+lemma integrableOn_iff_ofReal {X : Type*} [MeasurableSpace X] {μ : Measure X} {s : Set X}
+    {f : X → ℝ} : IntegrableOn f s μ ↔ IntegrableOn (fun x ↦ (f x : ℂ)) s μ := integrable_iff_ofReal
+
+lemma BetaIntegral_ofReal_interval {a b : ℝ} (ha : 0 < a) (hb : 0 < b) :
+    (∫ (x : ℝ) in (0)..1, (x : ℂ) ^ (↑a - 1:ℂ) * (1 - ↑x) ^ (↑b - 1:ℂ)).re =
+      ∫ (x : ℝ) in (0)..1, x ^ (a - 1) * (1 - x) ^ (b - 1) := by
+  rw [intervalIntegral.integral_of_le (by norm_num), intervalIntegral.integral_of_le (by norm_num),
+    ← Complex.setIntegral_re]
+  refine setIntegral_congr (measurableSet_Ioc) fun x hx ↦ ?_
+  rw [mem_Ioc] at hx
+  norm_cast
+  rw [← Complex.ofReal_cpow, ← Complex.ofReal_cpow]
+  simp
+  any_goals linarith
+  have ha' : 0 < (a: ℂ).re := by
+    rw [Complex.ofReal_re]
+    exact ha
+  have hb' : 0 < (b : ℂ).re := by
+    rw [Complex.ofReal_re]
+    exact hb
+
+  rw [← intervalIntegrable_iff_integrableOn_Ioc_of_le (by norm_num)]
+  exact Complex.betaIntegral_convergent ha' hb'
+
+lemma BetaIntegral_ofReal {a b : ℝ} (ha : 0 < a) (hb : 0 < b) :
+    Beta a b = ∫ (x : ℝ) in Ioo 0 1, x ^ (a - 1) * (1 - x) ^ (b - 1) := by
```

Comment:
Why don't you write it as an integral in `(0)..1`?

Reply:
fixed